android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudRepository.java

```diff
@@ -123,9 +125,18 @@ public void invalidateLexicalModelCache(@NonNull Context context) {
   {
     String deviceType = CloudDataJsonUtil.getDeviceTypeForCloudQuery(aContext);
 
+    // Sanitize appVersion to #.#.# to match the API spec
+    // Regex needs to match the entire string
+    String appVersion = BuildConfig.VERSION_NAME;
+    Pattern pattern = Pattern.compile("(\\d+\\.\\d+\\.\\d+).*");
```

Comment:
Am I right in thinking that this could match anywhere in the string? Suggest adding a `^` to force start-of-string match.
```suggestion
    Pattern pattern = Pattern.compile("^(\\d+\\.\\d+\\.\\d+)");
```

Reply:
The Matcher [documentation](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html) says it will attempt to match the entire string with the Pattern. But I've gone ahead and added the starting anchor.

Reply:
Yes, I read that and thought it was ambiguous -- after all, a regex pattern match is always done against the entire string... So I decided to learn: https://code.sololearn.com/cVRUy2BwauK8 and you were right :) -- the `.*` is needed on the end and the `^` is not needed on the front. Ah consistency!